### PR TITLE
Update invoke_rust_agent_expect_result to handle various exceptions

### PR DESCRIPTION
**Pull request by johnsonw**
Signed-off-by: Will Johnson <wjohnson@whamcloud.com>

### CODE REVIEW

File: chroma_core/lib/job.py [hunk 1]
```diff
@@ -200,7 +200,15 @@ def invoke_rust_agent(self, host, command, args={}):
     def invoke_rust_agent_expect_result(self, host, command, args={}):
         from chroma_core.services.job_scheduler.agent_rpc import AgentException
 
-        result = json.loads(self.invoke_rust_agent(host, command, args))
+        try:
```
Comment (jgrund) [inline chroma_core/lib/job.py]: I think the only thing inside this try should be 

`result = self.invoke_rust_agent(host, command, args)`

Then the json parsing should be in a separate try afterwards.

Comment (johnsonw) [reply chroma_core/lib/job.py]: sounds good 👍

File: chroma_core/lib/job.py [hunk 2]
```diff
@@ -200,7 +200,21 @@ def invoke_rust_agent(self, host, command, args={}):
...
+            result = self.invoke_rust_agent(host, command, args)
+        except RustAgentCancellation as e:
+            raise AgentException(host, command, args, "Cancelled: {}".format(e))
+        except ValueError as e:
```
Comment (johnsonw) [inline chroma_core/lib/job.py]: We definitely need the ValueError below when parsing json but i'm wondering if I need it here as well?

Comment (jgrund) [reply chroma_core/lib/job.py]: Nope.

File: chroma_core/lib/job.py [hunk 3]
```diff
@@ -200,7 +200,21 @@ def invoke_rust_agent(self, host, command, args={}):
...
+            raise AgentException(host, command, args, result)
+        except Exception as e:
+            raise AgentException(host, command, args, "Unexpected error: {}".format(e))
+
+        try:
+            result = json.loads(result)
+        except ValueError as e:
+            raise AgentException(host, command, args, "Error parsing json: {}".format(e))
+        except Exception as e:
+            raise AgentException(host, command, args, "Unexpected error while parsing json: {}".format(e))
```
Comment (jgrund) [inline chroma_core/lib/job.py]: Is there any other possible error besides a ValueError?

File: chroma_core/lib/job.py (hunk 2 above)
Comment (johnsonw) [reply chroma_core/lib/job.py]: ok great, i'll remove then.

Comment (jgrund) [reply chroma_core/lib/job.py]: To be clear, we shouldn't need the `ValueError` case.

File: chroma_core/lib/job.py (hunk 3 above)
Comment (johnsonw) [reply chroma_core/lib/job.py]: I think we just nee ValueError. https://github.com/simplejson/simplejson/blob/v3.0.5/simplejson/decoder.py#L33 shows the the encoder exception inherits from ValueError. Should I remove the general exception underneath?

Comment (jgrund) [reply chroma_core/lib/job.py]: Yes... unless there is a possibility Exception can be raised in addition to `ValueError`

Comment (johnsonw) [reply chroma_core/lib/job.py]: I did a quick search and din't see anything.

**Pull request merged**